### PR TITLE
When hitting the "Reset" button, only clear cloned blueprints

### PR DESCRIPTION
**Pull request by emilk**
### What
* Closes https://github.com/rerun-io/rerun/issues/5788

### Checklist
* [ ] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [ ] I've included a screenshot or gif (if applicable)
* [ ] I have tested the web demo (if applicable):
  * Using newly built examples: [rerun.io/viewer](https://rerun.io/viewer/pr/{{pr.number}})
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/{{pr.number}}?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/{{pr.number}}?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [ ] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [ ] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/{{ pr.number }})
- [Docs preview](https://rerun.io/preview/{{ pr.commit }}/docs) 
- [Examples preview](https://rerun.io/preview/{{ pr.commit }}/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_viewer_context/src/store_hub.rs [hunk 1]
```diff
@@ -155,6 +155,19 @@ impl StoreHub {
         // If we have an app-id, then use it to look up the blueprint.
         let app_id = self.active_application_id.clone()?;
 
+        // Check that default and active blueprints exists,
+        // in case some of our book-keeping is broken.
```
Comment (Wumpf) [inline crates/re_viewer_context/src/store_hub.rs:159]: unclear to me when this would happen. Do we know why this is needed?

**Review by Wumpf — APPROVED** [review]
a bit hard to follow at times, but each of the individual changesets seems to make things better!

File: crates/re_viewer_context/src/store_hub.rs (hunk 1 above)
Comment (emilk) [reply crates/re_viewer_context/src/store_hub.rs:159]: It's not needed right now; it's defensive coding against future failures in book keeping.
